lib/dns.js

```diff
@@ -22,6 +22,8 @@
 'use strict';
 
 const cares = internalBinding('cares_wrap');
+const { toASCII } = internalBinding('config').hasIntl ?
+  internalBinding('icu') : require('punycode');
```

Comment:
What's the policy here, considering `punycode` is deprecated?

Reply:
I don't know :man_shrugging:

Reply:
FWIW, we do the same thing in `lib/url.js`, so it's probably fine.

Reply:
Using punycode is the fallback if Node.js is compiled without ICU.

Reply:
I think it's only deprecated for userland.